src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/ApAreaEntityFactory.kt

```diff
@@ -8,8 +8,10 @@ import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomStringUpperCa
 import java.util.UUID
 
 class ApAreaEntityFactory(
-  apAreaTestRepository: ApAreaTestRepository
+  apAreaTestRepository: ApAreaTestRepository?
 ) : PersistedFactory<ApAreaEntity, UUID>(apAreaTestRepository) {
```

Comment:
I dunno if this is legit Kotlin, but if we had two factory types could set the return type like this:

```suggestion
 ) : apAreaTestRepository ? PersistedFactory<ApAreaEntity, UUID>(apAreaTestRepository) : Factory<ApAreaEntity, UUID>(apAreaTestRepository) {
```